### PR TITLE
Proposal rules

### PR DESCRIPTION
**Pull request by mvr320**

### CODE REVIEW

File: content/rules.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# Rules
+
+These rules and the chosen format might change, when the measures for the pandemic change.
+
+# General Rules
+
+## Definitions
+
+- BAPC: The Benelux Algorithm Programming Contest 2021. Hosted by the Vrije Universiteit Amsterdam.
+- Organization: the members of the organizing committee.
```
Comment (shanemin) [inline content/rules.md:16]:
```suggestion
- Organization: The members of the organizing committee.
```

File: content/rules.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Participation
+
+### Introduction
+
+- Participation should be done in teams of 3 persons.
+- There are two team classifications: one for student teams and one for business teams.
+- The organization can decide to change classifications before the start of the contest.
+- Changing the composition of a team is possible after approval by the organization.
+- Changing the composition should be done before the start of the contest the substitutions should be registered as a reserve.
```
Comment (shanemin) [inline content/rules.md:33]:
```suggestion
- Changing the composition should be done before the start of the contest and substitutions should be registered as a reserve.
```

File: content/rules.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+- The organization decides the amount of teams for each institution allowed competing.
```
Comment (shanemin) [inline content/rules.md:34]:
```suggestion
- The organization decides the amount of teams allowed for each institution competing.
```

File: content/rules.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- The responsibility for listing which of its teams+reserves should be considered for participation at the BAPC lies with their institute.
```
Comment (shanemin) [inline content/rules.md:35]:
```suggestion
- The responsibility for listing which of its teams+reserves should be considered for participation at the BAPC lies within their respective institution.
```

File: content/rules.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+- The organization has the right to deny the participation of teams before the start of the contest.
+
+### Photos and videos
+
+There might be a public livestream of the award ceremony available on the website. The recording will be publicly available afterwards too. If you do not want to appear on this livestream or want to be removed from the footage afterwards, please contact the organization. The organization can decide to not grant this request.
```
Comment (shanemin) [inline content/rules.md:40]:
```suggestion
There may be a public livestream of the award ceremony available on the website. The recording will be made publicly available afterwards. If you do not want to appear on this livestream, or want to be removed from the footage afterwards, please contact the organization. The organization can decide to deny this request.
```

File: content/rules.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+### Student teams
+
+A student team:
+
+- Participates for free.
+- Exists of students from the same institution and who are not participating in another team. The institutions must be located in either Belgium, Luxembourg or The Netherlands.
```
Comment (shanemin) [inline content/rules.md:47]:
```suggestion
- Exists of students from the same institution that are not members of another team. The institutions must be located in either Belgium, Luxembourg or The Netherlands.
```

File: content/rules.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+### Introduction
+
+- The language of the contest is English. All written contest materials will be in English.
+- The contest lasts for 5 hours.
+- From the beginning until one hour before the end of the BAPC, the scores are displayed.
+
+### Problems
+
+- At least six problems will be posed. The jury intents to develop problems with as little as possible dependence on detailed knowledge of a particular applications area or a particular contest language.
```
Comment (shanemin) [inline content/rules.md:69]:
```suggestion
- At least six problems will be posed. The jury intends to develop problems with minimal dependence on detailed knowledge of a particular application area or a particular contest language.
```

File: content/rules.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+- A contestant may submit a claim of ambiguity or error in a problem statement by submitting a clarification request to a judge. If the judges agree that an ambiguity or error exists, a clarification will be issued to all contestants.
```
Comment (shanemin) [inline content/rules.md:70]:
```suggestion
- A contestant may submit a claim of ambiguity or error for a problem statement by submitting a clarification request to a judge. If the judges agree that an ambiguity or error exists, a clarification will be issued to all contestants.
```

File: content/rules.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+- The jury has the right to change or withdraw problems during the contest before informing all teams.
```
Comment (shanemin) [inline content/rules.md:71]:
```suggestion
- The jury has the right to change or withdraw problems during the contest prior to informing all teams.
```

File: content/rules.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+### System
+
+- A solution for a problem has to be written in C99, C++ (version 14 or later will be available), Python (versions 3 will be available), Java (version 11 or later) or Kotlin (version 1.3), unless the problem statement explicitly states otherwise. Business teams can request other submission languages, on which the organization will decide whether it will be made available for submission.
```
Comment (shanemin) [inline content/rules.md:75]: I think the Java version should be 8+...

File: content/rules.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+- A team is disqualified if they submit harmful code that disrupts the contest
```
Comment (shanemin) [inline content/rules.md:76]:
```suggestion
- A team is disqualified if they submit harmful code that disrupts the contest.
```

File: content/rules.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+- A submission can use the standard library of the language.
+- A list of usable libraries are provided by the organization before the contest.
+- On-site contest
+  - Each team will use at most one workstation.
+  - There is an explicit restriction on internet access during the contest. Interaction about problems is allowed via clarification requests with the jury and internal discussion within the team. Asking questions about the contest to crew or coaches is subject to disqualification.
+- Online contest
+  - Each team will use at most three workstations, i.e. at most one computer per team member.
+  - There is no explicit restriction on internet access during the contest. The organization discourages copying code from reference materials as solutions will be checked on similarities. If there is suspicion of plagiarism among teams, both teams are subject to disqualification.
```
Comment (shanemin) [inline content/rules.md:84]:
```suggestion
  - There is no explicit restriction on internet access during the contest. The organization discourages copying code from reference material, as solutions will be checked on similarities. If there is suspicion of plagiarism among teams, both teams are subject to disqualification.
```

File: content/rules.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+  - Interaction about problems is allowed from clarification requests with the jury and internal discussion within the team. Asking questions about the contest to outside parties such as sites such as Stack Overflow are not allowed. The same applies to general programming errors.
```
Comment (shanemin) [inline content/rules.md:85]:
```suggestion
  - Interaction about problems is allowed from clarification requests with the jury and internal discussion within the team. Asking questions about the problems to third-parties sources, such as Stack Overflow, is not allowed. The same applies to general programming errors.
```

File: content/rules.md [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+
+### Judgment
+
+- Each run is judged as accepted or rejected by a judge, and the team is notified of the results.
+- For each problem, the jury has a correct solution and test data.
```
Comment (shanemin) [inline content/rules.md:90]:
```suggestion
- For each problem, the jury has the correct solution and test data.
```

File: content/rules.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+- A problem is solved after acceptance by the judges. The judges are solely responsible for accepting or rejecting runs.
+- The winner of the pool is decided by (in order):
+  - The team with the most solved problems.
+  - The team with the least total time. The total time is the sum of the time elapsed from the beginning of the contest to the submission of the accepted run plus 20 penalty minutes for every rejected run for that problem regardless of the submission time. Time is not consumed when the problem is not solved.
```
Comment (shanemin) [inline content/rules.md:94]:
```suggestion
  - The team with the least total time. The total time is the sum of the time elapsed from the beginning of the contest to the submission of the accepted run, plus 20 penalty minutes for every rejected run for that problem regardless of the submission time. Time is not consumed when the problem is not solved.
```

File: content/rules.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+  - The team that first submitted its last accepted problem is ranked higher. When a tie still remains, the team that first submitted its second-last accepted problem is ranked higher, and so on. When this does not resolve the tie, the ranks will be determined by chance.
+- The problem set is the responsibility of the jury.
+- The jury must be contacted through the "clarification requests".
```
Comment (shanemin) [inline content/rules.md:97]:
```suggestion
- The jury must be contacted through a "clarification request".
```

File: content/rules.md [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+## Special Rules
+
+- A team may be disqualified by the organization or jury for any activity that jeopardizes the contest such as dislodging extension cords, unauthorized modification, unintended use of contest materials, using forbidden function or library calls or distracting behavior.
```
Comment (shanemin) [inline content/rules.md:101]:
```suggestion
- A team may be disqualified by the organization or jury for any activity that jeopardizes the contest, such as dislodging extension cords, unauthorized modification, unintended use of contest materials, using forbidden function or library calls, or distracting behavior.
```

File: content/rules.md [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+- When a team is disqualified the coach should directly be notified, the team will be notified by the organization after the contest has ended. Exceptions to this rule are when the team obstructs other teams in participating.
```
Comment (shanemin) [inline content/rules.md:102]:
```suggestion
- When a team is disqualified the coach should be notified directly. The team will be notified by the organization after the contest has ended. Exceptions to this rule are when the team obstructs other teams in participating.
```

File: content/rules.md [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+- The organization has the right to stop the contest, extending the contest time, temporarily block submissions for all teams or change the scores in exceptional conditions.
```
Comment (shanemin) [inline content/rules.md:103]:
```suggestion
- The organization has the right to stop the contest, extend the contest time, temporarily block submissions for all teams, or change the scores in exceptional conditions.
```

File: content/rules.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+- In situations to which no rule applies or there is ambiguity about the rules, the organization decides.
```
Comment (shanemin) [inline content/rules.md:104]:
```suggestion
- In situations to which no rule applies, or there is ambiguity about the rules, the organization decides the outcome.
```

File: content/rules.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+- If necessary, the organization can make changes and exceptions to the rules.
```
Comment (shanemin) [inline content/rules.md:105]:
```suggestion
- If necessary, the organization can make changes and exceptions to the aforementioned rules.
```

File: content/rules.md (hunk 2 above)
Comment (mvr320) [reply content/rules.md:33]: This needs rewrite from my side.

**Review by shanemin — COMMENTED** [review]
Grammar changes, my changes are not necessarily correct

File: content/rules.md [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+### Photos and videos
+
+There may be a public livestream of the award ceremony available on the website. The recording will be made publicly available afterwards. If you do not want to appear on this livestream, or want to be removed from the footage afterwards, please contact the organization. The organization can decide to deny this request.
+
+### Student teams
+
+A student team:
+
+- Participates for free.
+- Exists of students from the same institution that are not members of another team. The institutions must be located in either Belgium, Luxembourg or The Netherlands.
```
Comment (mvr320) [inline content/rules.md:47]: I assume we're fine with people being reserve for multiple teams?

File: content/rules.md (hunk 10 above)
Comment (mvr320) [reply content/rules.md:75]: I think I directly copied this section and should look at it again with real values. Might be smart to discuss this soon with the other sides.

File: content/rules.md (hunk 14 above)
Comment (mvr320) [reply content/rules.md:90]: Something can have multiple solutions (multiple strategies) so I think there is no highlander situation here.

**Review by EuiYeonJang — APPROVED** [review]
Much needed addition to the website!

File: content/rules.md (hunk 10 above)
Comment (mvr320) [reply content/rules.md]: When we build the image we will revisit this.

File: content/rules.md [hunk 23]
```diff
@@ -23,6 +23,81 @@ These rules and the chosen format may change depending on the measures for the C
...
+- The responsibility for listing which of its teams (and reserves) should be considered for participation at the BAPC lies within their respective institution.
+- The organization has the right to deny the participation of teams before the start of the contest.
+
+### Student teams
+
+A student team:
+
+- Participates for free.
+- Consists of students from the same institution that are not members of another team. The institution must be located in either Belgium, Luxembourg or The Netherlands.
+- Has a coach, which is the contact person of a team. This should be a student or faculty member of the institution.
```
Comment (EBaalhuis) [inline content/rules.md:46]: Has a coach, _who_

**Pull request closed without merging**